dot/digest/digest.go

```diff
@@ -225,6 +257,35 @@ func (h *Handler) handleBlockFinalisation(ctx context.Context) {
 	}
 }
 
+// setBABEDigestsOnFinalization is called only when a block is finalised
+// and defines the correct next epoch data and next config data.
+func (h *Handler) setBABEDigestsOnFinalization(finalizedHeader *types.Header) error {
+	currEpoch, err := h.epochState.GetEpochForBlock(finalizedHeader)
+	if err != nil {
+		return fmt.Errorf("cannot get epoch for block %d (%s): %w",
+			finalizedHeader.Number, finalizedHeader.Hash(), err)
+	}
+
+	nextEpoch := currEpoch + 1
+
+	err = h.epochState.FinalizeBABENextEpochData(nextEpoch)
+	if err != nil && !errors.Is(err, state.ErrEpochNotInMemory) {
+		return fmt.Errorf("cannot finalize babe next epoch data for block number %d (%s): %w",
+			finalizedHeader.Number, finalizedHeader.Hash(), err)
+	}
+
+	err = h.epochState.FinalizeBABENextConfigData(nextEpoch)
+	if err == nil {
+		return nil
+	} else if !errors.Is(err, state.ErrEpochNotInMemory) {
+		return fmt.Errorf("cannot finalize babe next config data for block number %d (%s): %w",
+			finalizedHeader.Number, finalizedHeader.Hash(), err)
+	}
+
+	// the epoch state does not contains any information about the next epoch
+	return ErrCannotDefineNextEpoch
```

Comment:
nit maybe flip these around

```suggestion
	} else if errors.Is(err, state.ErrEpochNotInMemory) {
		// the epoch state does not contains any information about the next epoch
		return ErrCannotDefineNextEpoch
	}

	return fmt.Errorf("cannot finalize babe next config data for block number %d (%s): %w",
		finalizedHeader.Number, finalizedHeader.Hash(), err)
```